app/src/main/java/org/stepik/android/view/step_quiz_fullscreen_code/ui/dialog/CodeStepQuizFullScreenDialogFragment.kt

```diff
@@ -0,0 +1,355 @@
+        centeredToolbar.setBackgroundColor(ContextCompat.getColor(requireContext(), R.color.new_primary_color))
+        centeredToolbar.setOnMenuItemClickListener { item ->
+            when (item?.itemId) {
+                android.R.id.home -> {
+                    true
+                }
+                R.id.action_reset_code -> {
+                    val dialog = ResetCodeDialogFragment.newInstance()
+                    if (!dialog.isAdded) {
+                        dialog.show(childFragmentManager, null)
+                    }
+                    true
+                }
+                else -> false
+            }
+        }
+
+        initViewPager()
+
+        val text = stepWrapper
+            .step
+            .block
+            ?.text
+            ?.takeIf(String::isNotEmpty)
+
+        if (text != null) {
+            instructionsLayout.stepQuizCodeTextContent.setText(text)
+            instructionsLayout.stepQuizCodeTextContent.setTextSize(16f)
+            instructionsLayout.stepQuizCodeTextContent.setTextIsSelectable(true)
+        }
+
+        val actionsListener = object : CoreCodeStepDelegate.ActionsListener {
+            override fun onChangeLanguageClicked() {
+                val dialog = ChangeCodeLanguageDialog.newInstance()
+                if (!dialog.isAdded) {
+                    dialog.show(childFragmentManager, null)
+                }
+            }
+
+            override fun onFullscreenClicked(lang: String, code: String) {}
+        }
+
+        if (savedInstanceState != null) {
+            lang = savedInstanceState.getString(LANG) ?: return
+            code = savedInstanceState.getString(CODE) ?: return
+        }
+
+        submitButtonSeparator = playgroundLayout.submitButtonSeparator
+        codeSubmitButton = playgroundLayout.codeSubmitButton
+        retryButton = playgroundLayout.stepQuizRetry
+        codeLayout = playgroundLayout.codeStepLayout
+
+        codeSubmitButton.setOnClickListener {
+            callback.onSyncCodeStateWithParent(lang, codeStepLayout.text.toString(), onSubmitClicked = true)
+            dismiss()
+        }
+        retryButton.changeVisibility(false)
+        setupCodeToolAdapter()
+        setupKeyboardExtension()
+
+        coreCodeStepDelegate = CoreCodeStepDelegate(
+            codeContainerView = playgroundLayout,
+            stepWrapper = stepWrapper,
+            codeQuizInstructionDelegate = CodeQuizInstructionDelegate(instructionsLayout, false),
+            actionsListener = actionsListener,
+            codeToolbarAdapter = codeToolbarAdapter
+        )
+
+        coreCodeStepDelegate.setLanguage(CodeStepQuizFormState.Lang(lang, code))
+        coreCodeStepDelegate.setDetailsContentData(lang)
+        fullScreenCodeViewPager.setCurrentItem(1, false)
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString(LANG, lang)
```

Comment:
Why `code` value is extracted from `codeStepLayout` but `lang` value is stored locally, maybe it will be better to not rely on syncing local properties and always use ones from `codeStepLayout`?